app/scripts/controllers/pod.js

```diff
@@ -87,6 +87,58 @@ angular.module('openshiftConsole')
       }
     };
 
+    /**
+     *  Will set the newTerm's isActive/isUsed to true, while deactivating the old's
+     */
+    $scope.onTerminalSelectChange = function(newTerm) {
+      // Deactivate all terminals
+      _.each($scope.containerTerminals, function(term) {
+        term.isActive = false;
+      });
+
+      newTerm.value.isActive = true;
+      newTerm.value.isUsed = true;
+    };
+
+    $scope.tabSelect = function() {
+      $scope.terminalTabWasSelected = true;
+    };
+
+    var makeTerminals = function(containerStatuses, existingTerms) {
+      var firstRun = false;
+
+      if(!existingTerms) {
+        existingTerms = {};
+        firstRun = true;
+      }
+
+      // keeping existing terminals, append new ones
+      // set the first one active if this is the first time called
+      _.each(containerStatuses, function(status, i) {
+        // using a hash map eliminates some looping & can get passed to
+        // view with a tweak to <ui-select choices="">
+        if(!existingTerms[status.name]) {
+          existingTerms[status.name] = {
+            name: status.name,
+            // on first run, if this is the first container, set it active
+            isActive: (firstRun && (i === 0)),
+            isUsed: (firstRun && (i === 0))
+          };
+        }
+      });
+
+      var defaultSelected = {
+        key: _.findKey($scope.containerTerminals, { 'isActive': true }),
+        value: _.find($scope.containerTerminals, { 'isActive': true })
+      };
+
+      if(defaultSelected) {
```

Comment:
can eliminate the if statement here, since you defined the object on 130 and know for certain it exists.